src/api/v1.py

```diff
@@ -55,24 +57,25 @@ def get_version():
 
 
 @v1.post("/job")
-def job_post():
+@v1.input(schemas.Job, location="json")
+@v1.output(schemas.JobId)
+def job_post(json_data: dict):
     """Add a job to the queue"""
     try:
-        data = request.get_json()
-        job_queue = data.get("job_queue")
+        job_queue = json_data.get("job_queue")
     except (AttributeError, BadRequest):
         # Set job_queue to None so we take the failure path below
         job_queue = ""
     if not job_queue:
         return "Invalid data or no job_queue specified\n", 400
 
     try:
-        job = job_builder(data)
+        job = job_builder(json_data)
     except ValueError:
-        return "Invalid job_id specified\n", 400
+        abort(400, message="Invalid job_id specified")
```

Comment:
fixed, and also added a unit test to cover the corner case where the queue could be specified but is empty. I also changed this to use 422 instead of 400 since that's a little more specific.